File: Advanced Methods of Programming/JavaClientServer - Spring/Persistence/src/main/java/Repo/RepoAngajat.java
package Repo;

import Model.Angajat;
import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import java.sql.Connection;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class RepoAngajat implements IRepository<String, Angajat> {
    private JdbcUtils dbUtils;

    private static final Logger logger= LogManager.getLogger();

    public RepoAngajat(Properties props) {
        logger.info("Initializing RepoAngajat with properties: {} ",props);
        this.dbUtils = new JdbcUtils(props);
    }

    @Override
    public int size() {
        logger.traceEntry();
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select count(*) as SIZE from angajat")) {
            try(ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    logger.traceExit(result.getInt("SIZE"));
                    return result.getInt("SIZE");
                }
            }
        }catch(SQLException ex){
            logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        return 0;
    }

    @Override
    public Angajat findOne(String user) {
        logger.traceEntry("finding angajat with id {}",user);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select * from angajat where userAngajat=?")) {
            preStmt.setString(1,user);
            try(ResultSet result = preStmt.executeQuery()) {
                if (result.next()) {
                    Angajat a = new Angajat(result.getString("userAngajat"), result.getString("passwordAngajat"));
                    logger.traceExit(a);
                    return a;
                }
            }
        }catch(SQLException ex){
            logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        logger.traceExit("No angajat found with id{}",user);
        return null;
    }

    @Override
    public Iterable<Angajat> findAll() {
        logger.traceEntry();
        List<Angajat> lst = new ArrayList<>();
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("select * from angajat")) {
            try(ResultSet result = preStmt.executeQuery()) {
                while (result.next()) {
                    lst.add(new Angajat(result.getString("userAngajat"), result.getString("passwordAngajat")));
                }
            }
        }catch(SQLException ex){
            logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        logger.traceExit(lst);
        return lst;
    }

    public void save(Angajat angajat) {
        logger.traceEntry("saving anagajat{}",angajat);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("insert into angajat values (?,?)")){
            preStmt.setString(1,angajat.getId());
            preStmt.setString(2,angajat.getPasswordAngajat());
            preStmt.executeUpdate();
        }catch (SQLException ex){
            logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        logger.traceExit();
    }


    public void delete(String idEntity) {
        logger.traceEntry("deleting angajat with {}",idEntity);
        Connection con=dbUtils.getConnection();
        try(PreparedStatement preStmt=con.prepareStatement("delete from angajat where id=?")){
            preStmt.setString(1,idEntity);
            preStmt.executeUpdate();
        }catch (SQLException ex){
        logger.error(ex);
            System.out.println("Error DB "+ex);
        }
        logger.traceExit();
    }


    public void update(String idAngajat,Angajat angajat) {
        logger.traceEntry("updating angajat with {}",idAngajat);
        Connection con = dbUtils.getConnection();
        Angajat a=findOne(idAngajat);
        if(a==null){ return; }
        String newId=angajat.getId();
        String newPassword=angajat.getPasswordAngajat();

        try(PreparedStatement preStmt=con.prepareStatement("update Users"+
                " set password = ? "+
                " where user = ? ")){
            preStmt.setString(1,newPassword);
        }catch(SQLException ex) {
            logger.error(ex);
            System.out.println("Error DB angajat update " + ex);
        }
        logger.traceExit();
    }
}

File: Advanced Methods of Programming/Client-ServerCSharp/Server/Repository/RepoAngajat.cs
using System;
using System.Collections.Generic;
using System.Data;
using SharedDLL.Domain;
//using log4net;

namespace Server.Repo
{
    public class RepoAngajat : IRepository<string, Angajat>
    {
        //private static readonly ILog log = LogManager.GetLogger("SortingTaskDbRepository");

        IDictionary<String, string> props;
        public RepoAngajat(IDictionary<String, string> props)
        {
            //log.Info("Creating RepoAngajat");
            this.props = props;
        }

        public IEnumerable<Angajat> FindAll()
        {
            IDbConnection con = DBUtils.getConnection(props);
            IList<Angajat> angajati = new List<Angajat>();
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select userAngajat,passwordAngajat from angajat where userAngajat=@userAngajat";

                using (var dataR = comm.ExecuteReader())
                {
                    while (dataR.Read())
                    {
                        string userAngajat = dataR.GetString(0);
                        string password = dataR.GetString(1);
                        Angajat angajat = new Angajat(userAngajat, password);
                        angajati.Add(angajat);
                    }
                }
            }

            return angajati;
        }

        public Angajat FindOne(string userAngajat)
        {
            //log.InfoFormat("Entering findOne with value {0}", id);
            IDbConnection con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select userAngajat,passwordAngajat from angajat where userAngajat=@userAngajat";
                IDbDataParameter paramId = comm.CreateParameter();
                paramId.ParameterName = "@userAngajat";
                paramId.Value = userAngajat;
                comm.Parameters.Add(paramId);

                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        string password = dataR.GetString(1);
                        Angajat angajat = new Angajat(userAngajat, password);
                        //log.InfoFormat("Exiting findOne with value {0}", task);
                        return angajat;
                    }
                }
            }
            //log.InfoFormat("Exiting findOne with value {0}", null);
            return null;
        }

        public void Save(Angajat angajat)
        {
            var con = DBUtils.getConnection(props);

            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "insert into angajat  values (@user, @password)";
                var paramId = comm.CreateParameter();
                paramId.ParameterName = "@user";
                paramId.Value = angajat.GetId();
                comm.Parameters.Add(paramId);

                var paramPass = comm.CreateParameter();
                paramPass.ParameterName = "@password";
                paramPass.Value = angajat.GetPasswordAngajat();
                comm.Parameters.Add(paramPass);

                var result = comm.ExecuteNonQuery();
                if (result == 0)
                    throw new RepositoryException("Nu s-a adaugat niciun angajat !");
            }
        }

        public int Size()
        {
            IDbConnection con = DBUtils.getConnection(props);
            using (var comm = con.CreateCommand())
            {
                comm.CommandText = "select count(*) from angajat";
                using (var dataR = comm.ExecuteReader())
                {
                    if (dataR.Read())
                    {
                        //log.InfoFormat("Exiting findOne with value {0}", task);
                        return dataR.GetInt32(0);
                    }
                }
            }
            //log.InfoFormat("Exiting findOne with value {0}", null);
            return 0;
        }
    }

}
